pkg/tenantNamespace/namespaceManager.go

```diff
@@ -79,9 +79,9 @@ func (nm *tenantNamespaceManager) CreateNamespace(clusterID string) (ns *v1.Name
 	// exit with an error, and retry during the next iteration.
 	ns = &v1.Namespace{
 		ObjectMeta: metav1.ObjectMeta{
-			Name: strings.Join([]string{tenantNamespaceRoot, clusterID}, "-"),
+			Name: tenantNamespaceRoot + "-" + cluster.ClusterID,
```

Comment:
the same here (and in the other occurrences)

Reply:
Why do you think the Sprintf/strings.Join is better?

Reply:
For two main reasons:
1. it is the approach used in all liqo components
2. it's easier to maintain

You can read in [this post](https://tomasvotruba.com/blog/2018/10/11/hi-my-name-is-tom-conctat-vs-in-sprintf-vs-in-string-variables/) the discussion about different approaches (this is about php, but it is very similar for go)